### DISCUSSION TITLE
S3 backend for files

### DISCUSSION THREAD

**Discussion by CkuT** [jonaswinkler/paperless-ng#883] [Feature Requests]
Hi !

It would be useful for min.io/scaleway/aws/...  S3-like compatible storage

**Comment by anokfireball** [comment]
I am currently using such a setup. What you're proposing _could_ be natively integrated into NG, but it is probably out-of-scope for the project. Storage backends are assumed to be simple fuse mounts, so the easiest way to support virtually any backend (local, temporary, remote,... you name it) is to expose your target as a mount point and then point the `PAPERLESS_MEDIA_ROOT` to this directory. For S3-esque buckets, you have at least three options:
- https://github.com/s3fs-fuse/s3fs-fuse
- https://github.com/skoobe/riofs
- https://github.com/rclone/rclone

In my setup, I am mirroring a local directory to two different backends (S3 and ZFS) using rclone. The benefit I get from it is encrypted object storage under my full control, while simultaneously having the performance of locally cached files.

**Reply by jonaswinkler** [reply]
Certainly out of scope. If I were to add support for that, I'd also have to maintain it, fix bugs, documentation, etc. And I barely have the time to keep up with the project even in its current state :)

**Comment by iverona** [comment]
Hi! I'm sure this is a pretty basic question, but I cannot figure how to get it working. This is my configuration:

```
Summary
=======

Target folder: /home/iverona/paperless-ng
Consume folder: /home/iverona/paperless-ng/consume
Media folder: /home/iverona/iverona-paperless-bucket/media
Data folder: /home/iverona/iverona-paperless-bucket/data
```

but when starting the webserver container I get:

`Error response from daemon: error while creating mount source path '/home/iverona/iverona-paperless-bucket/data': mkdir /home/iverona/iverona-paperless-bucket: file exists`

Where `/home/iverona/iverona-paperless-bucket/` is an empty, fuse-mounted GCS bucket. What am I doing wrong? 

thanks!

**Reply by iverona** [reply]
Hi,

just for the record, I had to mount the fuse volume with the `allow_others` option. Now it's working.
